src/Node/Stream.purs

```diff
@@ -61,6 +66,16 @@ type Writable r = Stream (write :: Write | r)
 -- | A duplex (readable _and_ writable stream)
 type Duplex = Stream (read :: Read, write :: Write)
 
+type Data = Either String Buffer
+
+-- | Read a chunk of data into `Data`
+-- | This is a mere internal plumbing function used by functions reading data
+-- | from a stream.
+type ChunkReader = Foreign -> Data
+readChunk :: ChunkReader
+readChunk = readChunkImpl Left Right
+foreign import readChunkImpl :: (String -> Data) -> (Buffer -> Data) -> ChunkReader
```

Comment:
The reason for this is that there are lots and lots of functions of type `String -> Data` which could do all kinds of things, but only one of type `forall l r. l -> Either l r` (ignoring type system escape hatches).